File: Java-164/chapter.13/Lab14/NumberPattern.java
/*
Write a recursive method called printNumPattern() to output the following number pattern.

Given a positive integer as input (Ex: 12), subtract another positive integer (Ex: 3) continually until 0 or a negative value is reached, 
and then continually add the second integer until the first integer is again reached.

Ex. If the input is:

12
3

the output is:

12 9 6 3 0 3 6 9 12 

*/

import java.util.Scanner;

public class NumberPattern {

    static void printNumPattern(int num1, int num2)
    {
        if (num1 <= 0)
        {
            System.out.printf("%d ",num1);
        }
        else
        {
            System.out.printf("%d ",num1);
            printNumPattern((num1 - num2), num2);
            System.out.printf("%d ",num1);
        }
    }

    public static void main(String[] args) 
    {
        Scanner scnr = new Scanner(System.in);
        printNumPattern(scnr.nextInt(), scnr.nextInt());
        scnr.close();
    } 
}

File: CPP-165/week-2/lab/number_pattern/src/number_pattern.cpp
#include <iostream>

void print_num_pattern(int n1, int n2)
{
    if (n1 <= 1) std::cout << n1 << " ";
    else
    {
        std::cout << n1 << " ";
        print_num_pattern(n1 - n2, n2);
        std::cout << n1 << " ";
    }
    
}